db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -8003,6 +8063,35 @@ valueExpressionPrimary() throws StandardException :
     }
 }
 
+/*
+ * <A NAME="valueTupleElement">tupleValueElement</A>
+ */
+ValueTupleNode
+valueTupleElement(ValueNode tupleOrFirstValue) throws StandardException:
+{
+    ValueTupleNode  valueTuple;
+    ValueNode       newValue;
+}
+{
+    {
+        if (tupleOrFirstValue instanceof ValueTupleNode) {
+            valueTuple = (ValueTupleNode) tupleOrFirstValue;
+        } else {
+            valueTuple = (ValueTupleNode) nodeFactory.getNode(
+                                    C_NodeTypes.VALUE_TUPLE_NODE,
+                                    getContextManager());
+            valueTuple.addValueNode((ValueNode) tupleOrFirstValue);
+        }
+        newValue = valueExpression();
```

Comment:
@arnaud-splice Can you move the call of valueExpression() up. valueExpression() is part of the parser rule, although eventually it will be compiled into a function of the same name, I think it is better to keep it as a parser rule and put explicitly at the top as shown below, following the tradition of all other parser rules:
```
ValueTupleNode
valueTupleElement(ValueNode tupleOrFirstValue) throws StandardException:
{
    ValueTupleNode  valueTuple;
    ValueNode       newValue;
}
{
    newValue = valueExpression()
    {
        if (tupleOrFirstValue instanceof ValueTupleNode) {
            valueTuple = (ValueTupleNode) tupleOrFirstValue;
        } else {
            valueTuple = (ValueTupleNode) nodeFactory.getNode(
                                    C_NodeTypes.VALUE_TUPLE_NODE,
                                    getContextManager());
            valueTuple.addValueNode((ValueNode) tupleOrFirstValue);
        }

        if (newValue instanceof ValueTupleNode) {
            throw StandardException.newException(SQLState.LANG_SYNTAX_ERROR,
                "Nested tuple not supported");
        }
        valueTuple.addValueNode(newValue);
        return valueTuple;
    }
}
```

Reply:
Sure, that's a good idea!